autoscale-groups-api/src/main/java/fish/payara/extensions/autoscale/groups/Scales.java

```diff
@@ -37,23 +37,27 @@
  * only if the new code is made subject to such option by the copyright
  * holder.
  */
+
 package fish.payara.extensions.autoscale.groups;
 
-import javax.validation.Constraint;
+import org.glassfish.hk2.api.Metadata;
+
+import javax.inject.Qualifier;
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 /**
- * Validator annotation for checking that a {@link ScalingGroup} ConfigBean has a 1:1 relationship with a
- * {@link fish.payara.enterprise.config.serverbeans.DeploymentGroup}.
+ * Qualifier annotation used to help link a {@link Scaler} service with a {@link ScalingGroup}.
+ *
+ * @author Andrew Pielage
  */
-@Target({ElementType.METHOD, ElementType.FIELD})
+@Qualifier
 @Retention(RetentionPolicy.RUNTIME)
-@Constraint(validatedBy= NotDuplicateDeploymentGroupTargetValidator.class)
-public @interface NotDuplicateDeploymentGroupTarget {
+@Target(ElementType.TYPE)
+public @interface Scales {
```

Comment:
The interface looks fine, but its name confuses me. What is its purpose? To mark which services should be linked to a scaling operation? Shouldn't `Scalable` or `ScalableService` be used instead?

Reply:
It's a qualifier that links a `Scaler` service to a `ScalingGroup` type.
It essentially adds metadata that helps to identify what type of `ScalingGroup` a particular `Scaler` service should be operating against.
It's used from the scale up and down admin commands, which are generic commands and therefore don't inherently know which service needs to work with each Scaling Group.

Reply:
The name is intended as "ServiceA _Scales_ ScalingGroupA"
An alternative would be to name it something like `ScalerFor`

Reply:
Fair enough, I think that `ScalerFor` is a better name since it clarifies the purpose of the annotation in a single glance.